### PR TITLE
[WIP] DustAPI() uses DUST_FRONT_API || url

### PR DESCRIPTION
**Pull request by lasryaric**
# Description

DustAPI() now uses `process.env.DUST_FRONT_API` by defaults, and accepts a `url` parameter to override it.
`DUST_FRONT_API` is set to `https://dust.tt` if null (in `types/[...]/dust_api.ts` only)

### CODE REVIEW

File: connectors/src/connectors/slack/bot.ts [hunk 1]
```diff
@@ -223,16 +223,13 @@ async function botAnswerMessage(
...
+    credentials: {
       workspaceId: connector.workspaceId,
       apiKey: connector.workspaceAPIKey,
     },
     logger,
-    {
-      useLocalInDev: true,
-    }
-  );
+  });
```
Comment (lasryaric) [inline connectors/src/connectors/slack/bot.ts:232]: in `connectors/`, we want to rely on `DUST_FRONT_API` all the time. We are using the `DustAPI` client like an external developer would use it.

**Review by spolu — CHANGES_REQUESTED** [review]
Left a few initial comments but I think we want to keep the default to be DUST_PROD_API and have that be used when there is no `url` passed.

In connectors we want to pass explicitely DUST_FRONT_API

I believe there should be no reference to DUST_FRONT_API outside of connectors.

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 2]
```diff
@@ -367,9 +367,7 @@ export async function* runDustApp(
   const prodCredentials = await prodAPICredentialsForOwner(owner, {
     useLocalInDev: true,
   });
-  const api = new DustAPI(prodCredentials, logger, {
-    useLocalInDev: true,
-  });
+  const api = new DustAPI({ credentials: prodCredentials, logger });
```
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/lib/api/assistant/actions/dust_app_run.ts:370]: In this implementation I completely removed the variable `useLocalInDev`. So not sure how to read this message.

**Comment by spolu** [comment]
As per slack thread to be revisited

**Pull request closed without merging**